service/storage_proxy.cc

```diff
@@ -273,7 +273,9 @@ class hint_mutation : public shared_mutation {
     }
     virtual future<> apply_locally(storage_proxy& sp, storage_proxy::clock_type::time_point timeout,
             tracing::trace_state_ptr tr_state) override {
-        return make_exception_future<>(std::runtime_error("Executing hint locally doesn't make sense"));
+        // A hint will be sent to all relevant endpoints when the endpoint it was originally intended for
+        // becomes unavailable - this might include the current node
+        return sp.mutate_hint(_schema, *_mutation, timeout);
```

Comment:
Nice catch!
But I think this deserves to be in a separate patch.

Reply:
Previously, `hint_mutation` would only be used when hint's destination node is in the cluster (namely, `storage_proxy::send_hint_to_endpoint`). In case destination node was no longer the cluster, the path that sent the hint to its new replicas used `shared_mutation`.

This change is not really a bugfix, because it makes `hint_mutation` implement a method that was not used before, but required by `mutation_holder` interface. I did not hit the "Executing hint locally doesn't make sense" exception when I was testing a node replacement scenario without this patch.

Reply:
Got it.